### PR TITLE
Replace micronaut-jackson-databind with micronaut-serialization in gcp modules.

### LINKED ISSUE #751: uncouple the module from jackson-databind and use serde instead
### Issue description

We are using in multiple instances 

`implementation(mn.micronaut.jackson.databind)`

### LINKED ISSUE #842: GCP common still depends on micronaut-jackson-databind
### Issue description

https://github.com/micronaut-projects/micronaut-gcp/blob/master/gcp-common/build.gradle#L7

### PR DESCRIPTION
**Pull request by wetted**
closes #842
closes #751

### CODE REVIEW

File: gcp-common/src/main/java/io/micronaut/discovery/cloud/gcp/GoogleComputeInstanceMetadataResolver.java [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
-import com.fasterxml.jackson.databind.ObjectMapper;
 import io.micronaut.context.annotation.Primary;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.env.Environment;
 import io.micronaut.core.util.StringUtils;
 import io.micronaut.discovery.cloud.ComputeInstanceMetadata;
 import io.micronaut.discovery.cloud.ComputeInstanceMetadataResolver;
 import io.micronaut.discovery.cloud.NetworkInterface;
+import io.micronaut.json.tree.JsonNode;
+import io.micronaut.serde.ObjectMapper;
```
Comment (graemerocher) [inline gcp-common/src/main/java/io/micronaut/discovery/cloud/gcp/GoogleComputeInstanceMetadataResolver.java:26]: think you can support both Jackson and Micronaut serialisation if you use `JsonMapper.createDefault()` instead

Comment (graemerocher) [reply gcp-common/src/main/java/io/micronaut/discovery/cloud/gcp/GoogleComputeInstanceMetadataResolver.java:26]: See https://github.com/micronaut-projects/micronaut-oracle-cloud/blob/510daa6a79bf2c25843e652bd7140260d85b5e92/oraclecloud-common/src/main/java/io/micronaut/discovery/cloud/oraclecloud/OracleCloudMetadataResolver.java

**Comment by wetted** [comment]
The `gcp-logging` and `gcp-tracing` are disabled modules, so I did not touch them., since I don't know the plan for them. If `gcp-logging` is added back then it needs to have the `jackson-databind` dependency addressed.

File: gcp-function-http/src/main/java/io/micronaut/gcp/function/http/jackson/GoogleJacksonConfiguration.java [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-import jakarta.inject.Singleton;
-
-/**
- * Disables module scan for Jackson which is slow in function context.
- *
- * @author graemerocher
- * @since 1.2.0
- */
-@Singleton
-public class GoogleJacksonConfiguration implements BeanCreatedEventListener<JacksonConfiguration> {
```
Comment (sdelamo) [inline gcp-function-http/src/main/java/io/micronaut/gcp/function/http/jackson/GoogleJacksonConfiguration.java:31]: should we keep this but annotate with `@Requires(classes = JacksonConfiguration.class)` and add micronaut.jackson as compileOnly

File: gcp-function-http/src/main/java/io/micronaut/gcp/function/http/jackson/ObjectMapperCustomizer.java [hunk 3]
```diff
@@ -1,42 +0,0 @@
...
-
-import jakarta.inject.Singleton;
-
-/**
- * Customizes the jackson object mapper to ensure essential modules are registered.
- *
- * @author graemerocher
- * @since 1.2.0
- */
-@Singleton
```
Comment (sdelamo) [inline gcp-function-http/src/main/java/io/micronaut/gcp/function/http/jackson/ObjectMapperCustomizer.java:32]: should we keep this but annotate with `@Requires(classes = JacksonConfiguration.class)` and add micronaut.jackson as `compileOnly`?

File: gcp-logging/src/main/java/io/micronaut/gcp/logging/StackdriverJsonLayout.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public StackdriverJsonLayout() {
         this.includeException = false;
         this.includeTraceId = true;
         this.includeSpanId = true;
-        ObjectMapper mapper = new ObjectMapper();
+        ObjectMapper mapper = ObjectMapper.getDefault();
```
Comment (sdelamo) [inline gcp-logging/src/main/java/io/micronaut/gcp/logging/StackdriverJsonLayout.java:57]: use `JsonMapper:createOrDefault` ?

File: gcp-logging/build.gradle [hunk 5]
```diff
@@ -6,8 +6,8 @@ dependencies {
     compileOnly(projects.micronautGcpTracing)
     api(projects.micronautGcpCommon)
     implementation(libs.logback.json.classic)
+    implementation(mnSerde.micronaut.serde.support)
```
Comment (sdelamo) [inline gcp-logging/build.gradle:9]: should this be `micronaut.micronaut.json` ?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: gcp-logging/src/main/java/io/micronaut/gcp/logging/StackdriverJsonLayout.java (hunk 4 above)
Comment (wetted) [reply gcp-logging/src/main/java/io/micronaut/gcp/logging/StackdriverJsonLayout.java:57]: I am making this change, but note that the `gcp-logging` module (as well as `gcp-tracing`) are still disabled in settings.gradle.

File: gcp-logging/build.gradle (hunk 5 above)
Comment (wetted) [reply gcp-logging/build.gradle:9]: I am making this change, but note that the gcp-logging module (as well as gcp-tracing) are still disabled in settings.gradle.

**Review by sdelamo — APPROVED** [review]

File: gcp-common/build.gradle [hunk 6]
```diff
@@ -3,8 +3,9 @@ plugins {
 }
 
 dependencies {
+    annotationProcessor(mnSerde.micronaut.serde.processor)
     compileOnly(mn.micronaut.discovery.core)
-    implementation(mn.micronaut.jackson.databind)
+    implementation(mnSerde.micronaut.serde.jackson)
```
Comment (yawkat) [inline gcp-common/build.gradle:8]: we shouldnt depend on any implementation

File: gcp-pubsub/build.gradle [hunk 7]
```diff
@@ -3,15 +3,18 @@ plugins {
 }
 
 dependencies {
+    annotationProcessor(mnSerde.micronaut.serde.processor)
+    implementation(mnSerde.micronaut.serde.support)
```
Comment (yawkat) [inline gcp-pubsub/build.gradle:7]: is this necessary?

**Review by yawkat — CHANGES_REQUESTED** [review]

File: gcp-pubsub/build.gradle (hunk 7 above)
Comment (sdelamo) [reply gcp-pubsub/build.gradle]: I have removed those https://github.com/micronaut-projects/micronaut-gcp/pull/848/commits/f4fbc94a86a8b93be0ad66f6e70459cc3abb1dc3

File: gcp-common/build.gradle (hunk 6 above)
Comment (sdelamo) [reply gcp-common/build.gradle]: changed here: https://github.com/micronaut-projects/micronaut-gcp/pull/848/commits/f4fbc94a86a8b93be0ad66f6e70459cc3abb1dc3

**Review by yawkat — APPROVED** [review]

**Pull request merged**